homeassistant/components/intent/timers.py

```diff
@@ -0,0 +1,816 @@
+            id=timer_id,
+            name=name,
+            start_hours=hours,
+            start_minutes=minutes,
+            start_seconds=seconds,
+            seconds=total_seconds,
+            language=language,
+            device_id=device_id,
+            created_at=created_at,
+            updated_at=created_at,
+        )
+
+        # Fill in area/floor info
+        device_registry = dr.async_get(self.hass)
+        if device_id and (device := device_registry.async_get(device_id)):
+            timer.area_id = device.area_id
+            area_registry = ar.async_get(self.hass)
+            if device.area_id and (
+                area := area_registry.async_get_area(device.area_id)
+            ):
+                timer.floor_id = area.floor_id
+
+        self.timers[timer_id] = timer
+        self.timer_tasks[timer_id] = self.hass.async_create_background_task(
+            self._wait_for_timer(timer_id, total_seconds, created_at),
+            name=f"Timer {timer_id}",
+        )
+
+        await asyncio.gather(
+            *(handler(TimerEventType.STARTED, timer) for handler in self.handlers)
+        )
+
+        _LOGGER.debug(
+            "Timer started: id=%s, name=%s, hours=%s, minutes=%s, seconds=%s, device_id=%s",
+            timer_id,
+            name,
+            hours,
+            minutes,
+            seconds,
+            device_id,
+        )
+
+        return timer_id
+
+    async def _wait_for_timer(
+        self, timer_id: str, seconds: int, updated_at: int
+    ) -> None:
+        """Sleep until timer is up. Timer is only finished if it hasn't been updated."""
+        try:
+            await asyncio.sleep(seconds)
+            if (timer := self.timers.get(timer_id)) and (
+                timer.updated_at == updated_at
+            ):
+                await self._timer_finished(timer_id)
+        except asyncio.CancelledError:
+            pass  # expected when timer is updated
+
+    async def cancel_timer(self, timer_id: str) -> None:
+        """Cancel a timer."""
+        timer = self.timers.pop(timer_id, None)
+        if timer is None:
+            raise TimerNotFoundError
+
+        if timer.is_active:
+            task = self.timer_tasks.pop(timer_id)
+            task.cancel()
+
+        timer.cancel()
+
+        await asyncio.gather(
+            *(handler(TimerEventType.CANCELLED, timer) for handler in self.handlers)
+        )
+
+        _LOGGER.debug(
+            "Timer cancelled: id=%s, name=%s, seconds_left=%s, device_id=%s",
+            timer_id,
+            timer.name,
+            timer.seconds_left,
+            timer.device_id,
+        )
+
+    async def add_time(self, timer_id: str, seconds: int) -> None:
+        """Add time to a timer."""
+        if seconds == 0:
+            # Don't bother cancelling and recreating the timer task
+            return
+
+        timer = self.timers.get(timer_id)
+        if timer is None:
+            raise TimerNotFoundError
```

Comment:
Fix adding 0 seconds to a non-existing timer not raising. 
```suggestion
        timer = self.timers.get(timer_id)
        if timer is None:
            raise TimerNotFoundError

        if seconds == 0:
            # Don't bother cancelling and recreating the timer task
            return
```